src/main/java/uk/gov/companieshouse/acsp/manage/users/repositories/AcspMembersRepository.java

```diff
@@ -10,18 +10,24 @@
 import uk.gov.companieshouse.acsp.manage.users.model.AcspMembersDao;
 import uk.gov.companieshouse.api.acsp_manage_users.model.AcspMembership.UserRoleEnum;
 
+import java.util.Optional;
+
 @Repository
 public interface AcspMembersRepository extends MongoRepository<AcspMembersDao, String> {
 
-    @Query( "{ 'acsp_number': ?0, 'user_role': { $in: ?1 }, 'user_id': { $regex: ?2 }, 'removed_by': ?3 }" )
-    Page<AcspMembersDao> findAllByAcspNumberUserRolesAndUserIdLike( final String acspNumber, final Set<UserRoleEnum> userRoles, final String userId, final String removed_by, final Pageable pageable );
+  @Query( "{ 'acsp_number': ?0, 'user_role': { $in: ?1 }, 'user_id': { $regex: ?2 }, 'removed_by': ?3 }" )
+  Page<AcspMembersDao> findAllByAcspNumberUserRolesAndUserIdLike( final String acspNumber, final Set<UserRoleEnum> userRoles, final String userId, final String removed_by, final Pageable pageable );
+
+  @Query( "{ 'acsp_number': ?0, 'user_role': { $in: ?1 }, 'user_id': { $regex: ?2 } }" )
+  Page<AcspMembersDao> findAllByAcspNumberUserRolesAndUserIdLike( final String acspNumber, final Set<UserRoleEnum> userRoles, final String userId, final Pageable pageable );
 
-    @Query( "{ 'acsp_number': ?0, 'user_role': { $in: ?1 }, 'user_id': { $regex: ?2 } }" )
-    Page<AcspMembersDao> findAllByAcspNumberUserRolesAndUserIdLike( final String acspNumber, final Set<UserRoleEnum> userRoles, final String userId, final Pageable pageable );
+  @Query(value = "{ 'user_id': ?0 }")
+  List<AcspMembersDao> fetchAllAcspMembersByUserId(final String userId);
 
-    @Query(value = "{ 'user_id': ?0 }")
-    List<AcspMembersDao> fetchAllAcspMembersByUserId(final String userId);
+  @Query(value = "{ 'user_id': ?0, 'removed_by': { $exists: false } }")
+  List<AcspMembersDao> fetchActiveAcspMembersByUserId(final String userId);
 
-    @Query(value = "{ 'user_id': ?0, 'removed_by': { $exists: false } }")
-    List<AcspMembersDao> fetchActiveAcspMembersByUserId(final String userId);
+  @Query(value = "{ 'acsp_number': ?0, 'user_id': ?1 }")
```

Comment:
with mongo repository you dont need to write a query you can have something like
AcspMemberDao fetchAllForAcspNumberAndUserIdAndRemovedAtIsNull that should get you the value.
you should be able to test it using the integrationtest.